src/System.Text.Json/src/System/Text/Json/Node/JsonNumber.cs

```diff
@@ -0,0 +1,610 @@
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonNumber(byte value) => new JsonNumber(value);
+
+        /// <summary>
+        ///   Converts an Int16 to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonNumber(short value) => new JsonNumber(value);
+
+        /// <summary>
+        ///   Converts an Int32 to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonNumber(int value) => new JsonNumber(value);
+
+        /// <summary>
+        ///   Converts an Int64 to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonNumber(long value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a Single to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonNumber(float value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a Double to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonNumber(double value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a SByte to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        [CLSCompliant(false)]
+        public static implicit operator JsonNumber(sbyte value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a UInt16 to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        [CLSCompliant(false)]
+        public static implicit operator JsonNumber(ushort value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a UInt32 to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        [CLSCompliant(false)]
+        public static implicit operator JsonNumber(uint value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a UInt64 to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        [CLSCompliant(false)]
+        public static implicit operator JsonNumber(ulong value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a Decimal to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonNumber(decimal value) => new JsonNumber(value);
+
+        /// <summary>
+        ///   Compares <paramref name="obj"/> to the numeric value of this instance. 
```

Comment:
This doesn't seem accurate since 1.1e1 and 11 are equal (numerically), but I believe the following would return false:
```C#
var number = new JsonNumber("1.1e1")
object anotherForm = (object)new JsonNumber("11");
return number.Equals(anotherForm); // false?
```

Reply:
Right, I will change it.